dev-tools/xtask/src/a4x2_deploy.rs

```diff
@@ -0,0 +1,472 @@
+        // differently in that case (no cleanup, use /work)
+        let in_ci = env::var("CI").is_ok();
+
+        let a4x2_package_tar = match &args.command {
+            // Unused by stop command, so harmless to fill in with a default
+            DeployCommand::Stop => Utf8PathBuf::from(DEFAULT_A4X2_PKG_PATH),
+
+            DeployCommand::Start(args) => args.package.clone(),
+            DeployCommand::RunTests(args) => args.package.clone(),
+        };
+        let a4x2_package_tar = a4x2_package_tar.canonicalize_utf8()?;
+
+        let home_dir = Utf8PathBuf::from(env::var("HOME")?);
+        let work_dir = if in_ci {
+            Utf8PathBuf::from("/work/a4x2-deploy")
+        } else {
+            home_dir.join(".cache/a4x2-deploy")
+        };
+
+        // a4x2 dir. will be created by the unpack step
+        let a4x2_dir = work_dir.join("a4x2-package-out");
+
+        // Output. Maybe in CI we want this to be /out
+        let out_dir = work_dir.join("a4x2-deploy-out");
+
+        Environment { a4x2_package_tar, work_dir, a4x2_dir, out_dir, in_ci }
+    };
+
+    match args.command {
+        DeployCommand::Stop => teardown_a4x2(&sh, &env)?,
+        DeployCommand::Start(_) | DeployCommand::RunTests(_) => {
+            if !env.in_ci {
+                // Teardown previous deploy if it exists, before wiping the data
+                // for it. If this errors, we assume the deploy doesn't exist
+                // and carry on.
+                let result = teardown_a4x2(&sh, &env);
+                eprintln!("teardown result: {:?}", result);
+                eprintln!("continuing regardless of whether there were errors");
+
+                // Delete any results from previous runs. We don't mind if
+                // there's errors. This needs to run as root in case there are
+                // artifacts owned by root left around from the deploy.
+                cmd!(sh, "pfexec rm -rf").arg(&env.work_dir).run()?;
+            }
+
+            // Create work dir
+            fs::create_dir_all(&env.out_dir)?;
+            fs::create_dir_all(&env.work_dir)?;
+            sh.change_dir(&env.work_dir);
+
+            unpack_a4x2(&sh, &env)?;
+            prepare_to_launch_a4x2(&sh, &env)?;
+
+            // Launch a4x2. This can fail. In fact it fails quite a bit...
+            // enough that perhaps it's owed at least one retry? But we do not
+            // retry right now.
+            //
+            // We capture the error so we can tear down a4x2 gracefully. Note
+            // that in the case of running tests, we will produce an error if
+            // launching a4x2 succeeds but the tests themselves fail.
+            let result =
+                try_launch_a4x2(&sh, &env).and_then(|_| match &args.command {
+                    DeployCommand::RunTests(test_args) => {
+                        run_tests(&sh, &env, &test_args)
+                    }
+                    _ => Ok(()),
+                });
```

Comment:
There are two sources of errors here: launching a4x2, and running tests. Could you add contexts for each error?